src/Magnum/Math/BoolVector.h

```diff
@@ -303,6 +304,25 @@ template<std::size_t size> inline BoolVector<size> BoolVector<size>::operator~()
     return out;
 }
 
+/* Specialization of helper types*/
+#if !defined(DOXYGEN_GENERATING_OUTPUT)
+template<std::size_t size>
+struct StrictWeakOrdering<BoolVector<size>> {
+    bool operator()(const BoolVector<size>& a, const BoolVector<size>& b) const {
+        auto ad = a.data();
+        auto bd = b.data();
+        for (std::size_t i = 0; i < BoolVector<size>::DataSize; ++i) {
+            if (ad[i] < bd[i])
+                return true;
+            else if (ad[i] > bd[i])
+                return false;
+        }
+
+        return false;
+    }
```

Comment:
Ugh, the mask I suggested is not correct, sorry. Correct version should be, I *think* (please correct me if I'm wrong again):

```cpp
const UnsignedByte mask = (1 << (size % 8)) - 1;
```

This error also didn't cause any test to fail, which points to the importance of extending the test to verify random padding values (as I mentioned above). Can you extend it? Thank you!

To make this slightly shorter, I think it could get simplified to a single loop, it loses `constexpr` but this code usually isn't in the critical path anyway:

```cpp
        for(std::size_t i = 0; i < BoolVector<size>::DataSize; ++i) {
            const UnsignedByte mask = (1 << (i + 1 < BoolVector<size>::DataSize ? 8 : size % 8)) - 1;
            if((ad[i] & mask) < (bd[i] & mask))
                return true;
            if((ad[i] & mask) > (bd[i] & mask))
                return false;
        }
```

Reply:
oops :scream:  You're right
I guess I was sleeping last night

Reply:
Fixed with my own calculation: `0xFF >> (BoolVector<size>::DataSize * 8 - size)`

`BoolVector<size>::DataSize * 8 - size` is the padding, so with this we just pad an all-ones-in-binary accordingly